apps/src/levelbuilder/ai-iteration-tools/FreeResponseDatasetMaker.tsx

```diff
@@ -0,0 +1,172 @@
+import Papa from 'papaparse';
+import React, {useState} from 'react';
+
+import {evaluateStudentWork} from '@cdo/apps/aiEvaluation/evaluationApi';
+
+import {fetchFreeResponseAnswers} from './StudentWorkSamplesApi';
+
+interface StudentFreeResponseAnswer {
+  studentId: number;
+  studentDisplayName: string;
+  sectionId: number;
+  studentWork: string;
+  question: string;
+}
+
+interface StudentFreeResponseEvaluation extends StudentFreeResponseAnswer {
+  aiEvaluation: string;
+  aiReasoning: string;
+  evaluationCriteria?: string;
+}
+
+const FreeResponseDatasetMaker: React.FC = () => {
+  const [datasetName, setDatasetName] = useState<string>('');
+  const [levelId, setLevelId] = useState<string>('');
+  const [unitId, setUnitId] = useState<string>('');
+  const [numSamples, setNumSamples] = useState<string>('25');
+  const [includeAiEvaluations, setIncludeAiEvaluations] =
+    useState<boolean>(false);
+  const [pending, setPending] = useState<boolean>(false);
+  const [fetchedSamples, setFetchedSamples] = useState<
+    StudentFreeResponseAnswer[]
+  >([]);
+  const [evaluatedSamples, setEvaluatedSamples] = useState<
+    StudentFreeResponseEvaluation[]
+  >([]);
+
+  const downloadCSV = () => {
+    const csv = Papa.unparse(evaluatedSamples);
+    const csvData = new Blob([csv], {type: 'text/csv;charset=utf-8;'});
+    const csvURL = window.URL.createObjectURL(csvData);
+    const tempLink = document.createElement('a');
+    tempLink.href = csvURL;
+    tempLink.setAttribute('download', datasetName);
+    tempLink.click();
+  };
+
+  const getStudentFreeResponseAnswers = async () => {
+    setPending(true);
+    const studentWorkRequest = {
+      includeAiEvaluations: includeAiEvaluations,
+      numSamples: Number(numSamples),
+      unitId: Number(unitId),
+      levelId: Number(levelId),
+    };
+    const studentAnswers = await fetchFreeResponseAnswers(studentWorkRequest);
+    setFetchedSamples(studentAnswers as unknown as StudentFreeResponseAnswer[]);
+    setPending(false);
+  };
+
+  // TODO: Evaluate the fetched student responses with AI
+
+  const basePrompt =
+    'You are a teaching assistant for a high school AP Computer Science class where the students are learning JavaScript.';
+
+  const getAIEvaluations = async () => {
+    setPending(true);
+    const responsePromises = fetchedSamples.map(async studentResponse => {
+      return evaluateStudentResponse(studentResponse);
+    });
+    await Promise.allSettled(responsePromises);
+  };
+
+  const evaluateStudentResponse = async (
+    studentAnswer: StudentFreeResponseAnswer
+  ) => {
+    const systemPrompt = `${basePrompt} Please review the student's work. Respond in correctly formatted JSON.
+        evaluationCriteria should just be a copy of "hello".
+        aiEvaluation should be your assessment of the student's work. Respond with "great", "ok", or "needs revision".
+        aiReasoning should be one sentence with your reasoning.`;
```

Comment:
use the backend system prompt when https://github.com/code-dot-org/code-dot-org/pull/64671 is in